### PR TITLE
tests to OSCAP Policy entity - Airgun

### PR DESCRIPTION
**Pull request by tstrych**
test results 
```
============================= test session starts ==============================
platform linux -- Python 3.6.6, pytest-3.4.0, py-1.5.3, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/tstrych/Documents/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collected 4 items2018-09-04 18:09:09 - conftest - DEBUG - Collected 4 test cases


test_oscappolicy.py                                                  [100%]

============================== 0 tests deselected ==============================
========================== 4 passed in 447.02 seconds ==========================
```

### CODE REVIEW

File: tests/foreman/ui_airgun/test_oscappolicy.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        policy_val = session.oscappolicy.read(name)
+        assert (policy_val['scap_content']
+                ['tailoring_file'] == tailoring_file_name)
+        assert policy_val['schedule']['period'] == 'Custom'
+        assert (policy_val['schedule']
+                ['period_selection']['cron_line'] == cron_line)
+
+
+@tier2
+def test_positive_check_dashboard(session, module_host_group, module_loc,
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_oscappolicy.py]: we are talking about dashboard here, so you can't just ignore:
```
self.assertEqual(self.dashboard.get_total_hosts_count(), 1)
            self.assertEqual(
                self.oscappolicy.wait_until_element(
                    locators[
                        'dashboard.hcc.hosts_percentage'
                    ] % 'Not audited').text,
                '100%'
            )
```

Functionality is not ready from framework point of view, but should be ready in a few days

Comment (tstrych) [reply tests/foreman/ui_airgun/test_oscappolicy.py]: I discuss with @abalakh, original test wasn't actually checking dashboard. But still I should use more from dashboard functionality. 

Here is how I change the `view/oscappolicy.py` I will push it if you like it. ( I was not sure how I have to push it if the PR is already closed) 
```
class SCAPPolicyDetailsView(BaseLoggedInView):
    title = Text('h1[text()[contains(., "Compliance policy")]]')

    @property
    def is_displayed(self):
        return self.browser.wait_for_element(
            self.title, exception=False) is not None

    @View.nested
    class HostsBreakdownStatus(View):
        ROOT = ".//li[@data-name='Status table']"
        status_list = ItemValueList()
        total_count = TotalCount()

    @View.nested
    class HostBreakdownChart(View):
        ROOT = ".//li[@data-name='Status chart']"
        hosts_breakdown = Text('//div[@id="overview"]/span[@id="pieLabel3"]')
```

I also want to use PieChart from Widgets, but xpath to chart_title_text and chart_title_value in this example is different as in widgets.

Comment (oshtaier) [reply tests/foreman/ui_airgun/test_oscappolicy.py]: so you need other dashboard widgets to check? Create new PR to airgun, you don't need to perform anything extra with already closed ones

Comment (abalakh) [reply tests/foreman/ui_airgun/test_oscappolicy.py]: @tstrych please proceed with your approach, no need to introduce new widget or inherit from piechart as turns out the widget is only visually similar and internally it's totally different. If something needs to be done in airgun - just rise new PR with changes, there's nothing you should do with closed PRs.

**Comment by tstrych** [comment]
I was not able to run all test at the same time, because test time is over 15 minute on my virtual machine, and  I always end with `NavigationTriesExceeded Error` or `HTTP: 500 error, so |I will provide test results, one by one, maybe you can give me a tip how to solve this problems.

Together it's five test, 2 in ther first test result and rest in others.
First test result: (for test_positive_create and test_positive_create_with_missing_org_and_loc)
```
pytest -k test_positive_create  tests/foreman/ui_airgun/test_oscappolicy.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-3.4.0, py-1.5.3, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/tstrych/Documents/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collected 5 items                                                                                                                                                                                                 
2018-09-26 13:19:26 - conftest - DEBUG - Collected 5 test cases


tests/foreman/ui_airgun/test_oscappolicy.py ..                                                                                                                                                              [100%]

=============================================================================================== 3 tests deselected ================================================================================================
==================================================================================== 2 passed, 3 deselected in 419.37 seconds =====================================================================================
```
Second: test_positive_delete
```
pytest -k test_positive_delete  tests/foreman/ui_airgun/test_oscappolicy.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-3.4.0, py-1.5.3, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/tstrych/Documents/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collected 5 items                                                                                                                                                                                                 
2018-09-26 13:34:26 - conftest - DEBUG - Collected 5 test cases


tests/foreman/ui_airgun/test_oscappolicy.py .                                                                                                                                                               [100%]

=============================================================================================== 4 tests deselected ================================================================================================
==================================================================================== 1 passed, 4 deselected in 190.21 seconds =====================================================================================
```
Third: test_positive_update
```
pytest -k test_positive_update  tests/foreman/ui_airgun/test_oscappolicy.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-3.4.0, py-1.5.3, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/tstrych/Documents/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collected 5 items                                                                                                                                                                                                 
2018-09-26 13:38:57 - conftest - DEBUG - Collected 5 test cases


tests/foreman/ui_airgun/test_oscappolicy.py .                                                                                                                                                               [100%]

=============================================================================================== 4 tests deselected ================================================================================================
==================================================================================== 1 passed, 4 deselected in 275.22 seconds =====================================================================================
```
Fourth: test_positive_check_dashboard
```
pytest -k test_positive_check_dashboard tests/foreman/ui_airgun/test_oscappolicy.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-3.4.0, py-1.5.3, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/tstrych/Documents/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collected 5 items                                                                                                                                                                                                 
2018-09-26 13:48:47 - conftest - DEBUG - Collected 5 test cases


tests/foreman/ui_airgun/test_oscappolicy.py .                                                                                                                                                               [100%]

=============================================================================================== 4 tests deselected ================================================================================================
==================================================================================== 1 passed, 4 deselected in 165.66 seconds =====================================================================================
```

**Pull request merged**